sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -165,8 +165,20 @@ private[hive] trait HiveStrategies {
 
   object Scripts extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
-      case logical.ScriptTransformation(input, script, output, child) =>
-        ScriptTransformation(input, script, output, planLater(child))(hiveContext) :: Nil
+      case logical.ScriptTransformation(input, script, output, child,
```

Comment:
I think a better place to extract the schema (the `output`) is in `Analyzer`,  `HiveContext` should be able to create its own rules for that, instead of doing this in `Strategy`. Otherwise it probably fails in resolving the attributes.

e.g.: 
```
SELECT transform(key + 1, value) USING '/bin/cat' FROM src ORDER BY key, value`
```

sorry, I didn't test that, let me know if I am wrong.